File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellerfoods/service/BrandService.java
package com.pinyougou.sellerfoods.service;

import com.github.pagehelper.PageInfo;
import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;

public interface BrandService {
    //查询所有的品牌
    public List<TbBrand> findAll();

    //分页查询
    public PageInfo findPage(int pageNum, int pageSize);

    //添加品牌
    public void add(TbBrand brand);

    //根据id查找品牌
    public TbBrand findOne(Long id);

    //根据id查找品牌后，再修改品牌
    public void update(TbBrand brand);

    //根据ids删除品牌
    public void delete(Long[] ids);

    //条件查询品牌
    public PageInfo findByExample(TbBrand brand,int page,int size);
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brand_Service.js
//创建服务层,主要负责和后台进行数据交互
app.service("brandService",function ($http) {
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    }
    this.findPage = function (page, size) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + size);
    }
    this.findOne = function (id) {
        return $http.get('/brand/findOne.do?id=' + id);
    }
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    }
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    }
    this.remove = function (selectIds) {
        return $http.post("../brand/delete.do",selectIds);
    }
    this.search = function (page,size,ExampleEntity) {
        return $http.post('../brand/findByExample.do?page='+page+'&size='+size,ExampleEntity);
    }
    //查询所有的品牌类表
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }
})
